src/AdapterPluginManager.php

```diff
@@ -39,14 +40,53 @@ class AdapterPluginManager extends AbstractPluginManager
     ];
 
     protected $factories = [
-        Adapter\IgBinary::class     => InvokableFactory::class,
-        Adapter\Json::class         => InvokableFactory::class,
-        Adapter\MsgPack::class      => InvokableFactory::class,
-        Adapter\PhpCode::class      => InvokableFactory::class,
-        Adapter\PhpSerialize::class => InvokableFactory::class,
-        Adapter\PythonPickle::class => InvokableFactory::class,
-        Adapter\Wddx::class         => InvokableFactory::class
+        Adapter\IgBinary::class             => InvokableFactory::class,
+        'zendserializeradapterigbinary'     => InvokableFactory::class,
+        Adapter\Json::class                 => InvokableFactory::class,
+        'zendserializeradapterjson'         => InvokableFactory::class,
+        Adapter\MsgPack::class              => InvokableFactory::class,
+        'zendserializeradaptermsgpack'      => InvokableFactory::class,
+        Adapter\PhpCode::class              => InvokableFactory::class,
+        'zendserializeradapter\phpcode'     => InvokableFactory::class,
+        Adapter\PhpSerialize::class         => InvokableFactory::class,
+        'zendserializeradapterphpserialize' => InvokableFactory::class,
+        Adapter\PythonPickle::class         => InvokableFactory::class,
+        'zendserializeradapterpythonpickle' => InvokableFactory::class,
+        Adapter\Wddx::class                 => InvokableFactory::class,
+        'zendserializeradapterwddx'         => InvokableFactory::class
```

Comment:
No. You should have a combination of aliases and factories. InvokableFactory will not work with the above aliases; it assumes the requested name is the class name. What you should do is:

- map each FCQN for a plugin to an `InvokableFactory`
- map aliases for each plugin to the FQCN. Typically, you'll have multiple variants, but typically all will be in *short form*. You don't have to do normalized versions of the FQCN as you did above.